main.go

```diff
@@ -198,6 +201,19 @@ func runWithOptions(opts Options, out output) {
 	})
 }
 
+type noEnveloper interface {
+	WithoutEnvelopes() encoding.Serializer
+}
+
+// withTransportSerializer may modify the serializer for the transport used.
+// E.g. Thrift payloads are not enveloped when used with TChannel.
+func withTransportSerializer(p transport.Protocol, s encoding.Serializer) encoding.Serializer {
+	if p == transport.TChannel && s.Encoding() == encoding.Thrift {
+		s = s.(noEnveloper).WithoutEnvelopes()
```

Comment:
Yeah I figure if yab ever adds support for Thrift over raw sockets, it'll use envelopes too, so seems like envelopes are the expected way to use Thrift.